### PR TITLE
Query block syntax update

### PR DESCRIPTION
**Pull request by archiewood**
### Description
This addresses #485, adding more standard markdown syntax for our SQL query interface.
It is backwards compatible so

````
```orders
select * from orders
```
````

will still work, but now, so will

````
```sql orders
select * from orders
```
````

### Checklist
- [ ] For UI or styling changes, I have added a screenshot or gif showing before & after
- [ ] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

### CODE REVIEW

**Review by ItsMeBrianD — COMMENTED** [review]
This looks good to me but will likely have some fairly major conflict; I can rebase and try to fix or we can reimplement, looks like the core feature (not the whitespace) is pretty straightforward.

**Comment by archiewood** [comment]
Core feature is simple yep.

I'd suggest we reimplement given it's only a few lines of code.

Happy to look at this later on this week, or for you to.

**Review by ud3sh — APPROVED** [review]

**Comment by ItsMeBrianD** [comment]
I'm more than happy to take it on; I'll plan on touching this tomorrow

**Comment by ItsMeBrianD** [comment]
Needs to wait for #734, depends on the refactors in that branch

File: packages/preprocess/src/extract-queries/extract-queries.cjs [hunk 1]
```diff
@@ -34,71 +31,85 @@ const extractExternalQueries = (content, filename) => {
...
+            compiledQueryString: content,
+            inputQueryString: content,
+            compiled: false,
+            inline: false,
+          };
+        } catch {
+          console.warn(`Failed to load sql file ${source}`);
+        }
+      } else if (typeof source === "object") {
+        const usedKey = Object.keys(source)[0];
```
Comment (Winterhart) [inline packages/preprocess/src/extract-queries/extract-queries.cjs:83]: Should we add a check that we can't get back an empty array from `.keys`?

File: packages/preprocess/src/extract-queries/extract-queries.cjs [hunk 2]
```diff
@@ -34,71 +31,85 @@ const extractExternalQueries = (content, filename) => {
...
+          return {
+            id: usedKey.toLowerCase(),
+            compiledQueryString: content,
+            inputQueryString: content,
+            compiled: false,
+            inline: false,
+          };
+        } catch {
+          console.warn(`Failed to load sql file ${value}`);
+        }
```
Comment (Winterhart) [inline packages/preprocess/src/extract-queries/extract-queries.cjs:109]: Could it become a function I am seeing this block twice?

**Review by Winterhart — COMMENTED** [review]
Looks greats only have 2 questions 👍🏼

File: packages/preprocess/src/extract-queries/extract-queries.cjs [hunk 3]
```diff
@@ -2,15 +2,35 @@ const unified = require("unified");
...
+ * @returns Query
+ */
+const readFileToQuery = (source) => {
+  try {
+    const content = fs
+      .readFileSync(`./sources/${source}`)
+      .toString()
+      .trim();
+    return {
+      id: id.toLowerCase(),
```
Comment (Winterhart) [inline packages/preprocess/src/extract-queries/extract-queries.cjs:24]: you might have to pass `id` as a parameter? 🤔

File: packages/preprocess/src/extract-queries/extract-queries.cjs [hunk 4]
```diff
@@ -34,71 +54,56 @@ const extractExternalQueries = (content, filename) => {
...
-      if (Object.keys(source).length > 1) {
-        console.warn(`Source object has more than one key, this may lead to unintended behavior. Only ${usedKey}: ${value} will be imported.`)
+  return frontmatter.sources
+    .map((source) => {
+      if (typeof source === "string") {
+        if (!validateSource(source)) return false;
+        const id = source.split(".sql")[0].replace("/", "_").replace("\\", "_");
+        return readFileToQuery(source)
+      } else if (typeof source === "object") {
+        const usedKey = Object.keys(source)?.[0] ?? "";
```
Comment (Winterhart) [inline packages/preprocess/src/extract-queries/extract-queries.cjs:92]: 👍🏼

**Review by Winterhart — APPROVED** [review]
Thanks for the quick changes

**Pull request merged**